tasks/deployers/greeter.ts

```diff
@@ -5,9 +5,16 @@ import { Greeter, Greeter__factory } from "../../typechain";
 
 task("deploy:Greeter")
   .addParam("greeting", "Say hello, be nice")
-  .setAction(async function (taskArguments: TaskArguments, { ethers }) {
+  .setAction(async function (taskArguments: TaskArguments, { config, ethers, network, run }) {
     const greeterFactory: Greeter__factory = await ethers.getContractFactory("Greeter");
     const greeter: Greeter = <Greeter>await greeterFactory.deploy(taskArguments.greeting);
     await greeter.deployed();
     console.log("Greeter deployed to: ", greeter.address);
+
+    if (network.name !== "hardhat" && config.etherscan.apiKey) {
+      await run("verify:verify", {
```

Comment:
IMO, if we call verify on etherscan right after contract deployment it will be always failed due to etherscan need time to load the smart contract code either. Then we should delay some minutes before invoke verification.

Reply:
Hey, yes I researched about this and I'll make it wait for atleast 1-2mins after deployment so etherscan could do it's stuff in the meantime. Thank you for the correction!